internal/pkg/runtime/platform/metal/metal.go

```diff
@@ -38,6 +40,34 @@ func (m *Metal) Configuration() ([]byte, error) {
 		return nil, fmt.Errorf("no config option was found")
 	}
 
+	u, err := url.Parse(*option)
+	if err != nil {
+		return nil, err
+	}
+
+	values := u.Query()
+
+	if len(values) > 0 {
+		for key := range values {
+			switch key {
+			case "uuid":
+				s, err := smbios.New()
+				if err != nil {
+					return nil, err
+				}
+
+				uuid, err := s.SystemInformation().UUID()
+				if err != nil {
+					return nil, err
+				}
+
+				values.Set("uuid", uuid.String())
```

Comment:
does this actually work? my understanding was that it won't update `u` ?

Reply:
This value is not guaranteed to be filled, so I would caution against using it directly.

Reply:
I guess you need `u.RawQuery = values.Encode()` before you do `u.String()`